pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/ParametricRuleViolation.java

```diff
@@ -38,6 +42,11 @@
     protected String methodName = "";
     protected String variableName = "";
 
+    public ParametricRuleViolation(Rule theRule, RuleContext ctx, T node, String message, String snippet) {
```

Comment:
Nobody uses this constructor outside tests, so I don't think your feature works. Actually it looks like it's a real pain to support this as you'll have to change the RuleViolationFactory of every language. This is not a good idea since that has already been removed from pmd 7...